src/libs/costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -41,65 +41,76 @@
 #include <string>
 #include <algorithm>
 #include <vector>
-#include <tf2/convert.h>
-#include <tf2/utils.h>
 #include <tf2_geometry_msgs/tf2_geometry_msgs.h>
 
 using namespace std;
 
 namespace costmap_2d
 {
 
-void move_parameter(ros::NodeHandle& old_h, ros::NodeHandle& new_h, std::string name, bool should_delete = true)
+void move_parameter(rclcpp::Node::SharedPtr old_h, rclcpp::Node::SharedPtr new_h, std::string name,
+    bool should_delete = true)
 {
-  if (!old_h.hasParam(name))
+  auto parameters_client_old = std::make_shared<rclcpp::SyncParametersClient>(old_h);
```

Comment:
You don't need to make a parameters client. There is one built into the node that is exposed through the node interface.

This block could be something like
```c++
XmlRpc::XmlRpcValue value;
if (!old_h.get_parameter(name, value))
  return;
else
 new_h.set_parameters({rclcpp::Parameter(name, value)});
```

The delete part I still don't know how to do.

Reply:
I made this an issue in #119